pkg/virt-launcher/virtwrap/network/podnic.go

```diff
@@ -35,14 +35,15 @@ import (
 )
 
 type podNIC struct {
-	vmi              *v1.VirtualMachineInstance
-	podInterfaceName string
-	launcherPID      *int
-	vmiSpecIface     *v1.Interface
-	vmiSpecNetwork   *v1.Network
-	handler          netdriver.NetworkHandler
-	cacheFactory     cache.InterfaceCacheFactory
-	dhcpConfigurator *dhcpconfigurator.Configurator
+	vmi                           *v1.VirtualMachineInstance
+	podInterfaceName              string
+	launcherPID                   *int
+	vmiSpecIface                  *v1.Interface
+	vmiSpecNetwork                *v1.Network
+	handler                       netdriver.NetworkHandler
+	cacheFactory                  cache.InterfaceCacheFactory
+	dhcpConfigurator              *dhcpconfigurator.Configurator
+	podNetworkConfiguratorFactory func(*podNIC) (infraconfigurators.PodNetworkInfraConfigurator, error)
```

Comment:
Making a function a member of a struct is tricky since golang's equality definition does not account for functions.

Furthermore, why is the only the infra configurator mocked ? Shouldn't the libvirt spec generator also be mocked ?

Reply:
This no longer applies